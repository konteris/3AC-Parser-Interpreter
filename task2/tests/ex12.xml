<program name="Return Call Test">
  <tac opcode="CALL" order="1">
    <dst type="label">@function</dst>
  </tac>
  <tac opcode="PRINTLN" order="2">
    <src1 type="string">Returned from function</src1>
  </tac>
  <tac opcode="JUMP" order="3">
    <dst type="label">@end</dst>
  </tac>
  <tac opcode="LABEL" order="4">
    <dst type="label">@function</dst>
  </tac>
  <tac opcode="PUSH" order="5">
    <src1 type="integer">5</src1>
  </tac>
  <tac opcode="RETURN" order="6">
  </tac>
  <tac opcode="LABEL" order="7">
    <dst type="label">@end</dst>
  </tac>
  <tac opcode="POP" order="8">
    <dst type="variable">x</dst>
  </tac>
    <tac opcode="PRINTLN" order="9">
        <src1 type="variable">x</src1>
    </tac>
</program>