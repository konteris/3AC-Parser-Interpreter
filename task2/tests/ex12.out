Returned from function
5
